dashdblocal_notebooks/livy-notebooks/IoT_Demo_Pyspark.ipynb

```diff
@@ -0,0 +1,1246 @@
+    "\n",
+    "You will notice that there are only a few rows compared to the number of generated records. For example, with a timeout set to 10s, you will collect approximately 8 rows. With a timeout of 50s, there will be approximetely 30 etc."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database\n",
+      "Batch written to database"
+     ]
+    }
+   ],
+   "source": [
+    "tableName= \"USERDATA.IOT_EVENTS\"\n",
+    "savemode = 'overwrite'\n",
+    "\n",
+    "def processStream2(rdd):\n",
+    "    global savemode\n",
+    "    rdd2 = rdd.map(lambda x: x[1])\n",
+    "    df = sqlsc.read.json(rdd2, schema)\n",
+    "    df2=df.select([\"payload.temperature\", \\\n",
+    "               \"payload.tempOutside\", \\\n",
+    "               \"payload.powerProd\", \\\n",
+    "               \"payload.noiseLevel1\", \\\n",
+    "               \"payload.time\", \\\n",
+    "               \"deviceId\", \\\n",
+    "               \"deviceType\", \\\n",
+    "               \"eventType\"])\n",
+    "    \n",
+    "    \n",
+    "    df2.write.format(\"com.ibm.idax.spark.idaxsource\"). \\\n",
+    "            option(\"dbtable\", tableName ). \\\n",
+    "            option(\"allowAppend\", \"TRUE\"). \\\n",
+    "            mode(savemode). \\\n",
+    "            save()\n",
+    "    print(\"Batch written to database\")\n",
+    "    savemode ='append'\n",
+    "    \n",
+    "runStreamingContext(processStream2)\n",
+    "    \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "An error was encountered:\n",
+      "Invalid status code '404' from http://idax29.svl.ibm.com:8998/sessions/73 with error payload: {\"msg\":\"Session '73' not found.\"}\n"
```

Comment:
I think this cell should not contain an error message :-)

Reply:
Just tested it and I don't obtain an error message... it's a 'session not found' error so it's not directly related to the content of this cell.